app/uk/gov/hmrc/apipublisher/services/OasParserImpl.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.apipublisher.services
+
+import javax.inject.Singleton
+import io.swagger.v3.oas.models.OpenAPI
+import uk.gov.hmrc.ramltools.domain.Endpoint
+import uk.gov.hmrc.apipublisher.models.oas._
+import uk.gov.hmrc.ramltools.domain.QueryParam
+import uk.gov.hmrc.apipublisher.util.ApplicationLogger
+import scala.util.control.NonFatal
+
+@Singleton
+class OasParserImpl() extends OasVersionDefinitionService.OasParser with ApplicationLogger {
+     
+  def trimContext(context: Option[String])(urlPattern: String): String = {
+    context.fold(urlPattern)(c => {
```

Comment:
I guess there is a chance we will trim the wrong thing here, if context is 'abc' and the url is '/abcd/efg' it will get trimmed - maybe doesn't matter?

Reply:
The contexts are fixed to a list of about 13 different allowable ones that get validated somewhere so we should be fine.  However it's worth probably ensuring the / after the context in the replace.